up-l3/udiscovery/v3/service.adoc

```diff
@@ -0,0 +1,253 @@
+= uDiscovery Service
+:toc: preamble
+:sectnums:
+:source-highlighter: highlight.js
+:client-proto-ref: xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[udiscovery.proto]
+:replicator-proto-ref: xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[udiscovery_replicator.proto]
+
+
+The key words "*MUST*", "*MUST NOT*", "*REQUIRED*", "*SHALL*", "*SHALL NOT*", "*SHOULD*", "*SHOULD NOT*", "*RECOMMENDED*", "*MAY*", and "*OPTIONAL*" in this document are to be interpreted as described in https://www.rfc-editor.org/info/bcp14[IETF BCP14 (RFC2119 & RFC8174)]
+
+----
+SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+
+See the NOTICE file(s) distributed with this work for additional
+information regarding copyright ownership.
+
+This program and the accompanying materials are made available under
+the terms of the Apache License Version 2.0 which is available at
+https://www.apache.org/licenses/LICENSE-2.0
+ 
+SPDX-FileType: DOCUMENTATION
+SPDX-License-Identifier: Apache-2.0
+----
+
+== Overview
+
+UDiscovery provides the means for uEs to discover services (and their metadata), for local and remote services. Implementation of uDiscovery takes inspiration from Domain Name System (DNS) principles in that we store and cache queries and replicate local information to more centralized databases. The local uDiscovery service acts as the first point of contact for discovery and then the discovery service looks up at the domain level service for information about services across a domain (i.e. vehicle). The central instance of uDiscovery service  is the last point of contact for global searches of services.
+
+UDiscovery Service implements two sets of APIs:
+
+- *Client APIs ({client-proto-ref}):* Used by uEs to find services and their topic metadata, and for recursive lookups at the domain and central levels.
+
+- *Replication API ({replicator-proto-ref}):* Used Internally by uDiscovery services to add, update, or delete service metadata from local, domain, and/or central service instances.
+
+<<udiscovery-design>> below provides an overview of the design and use of the Client and Replication APIs to implement the uDiscovery business logic.
+
+.uProtocol Discovery Design
+[#udiscovery-design]
+image::design.drawio.svg[Overview]
+
+=== Terms
+
+- *Fully Qualified Service URI (FQURI):* A URI that contains all the information needed to uniquely identify a service. A fully qualified service URI contains the following information:
+  - *ue_id:* The unique identifier of the service (both the instance and service ID portions).
+  - *authority_name:* The name of the authority that hosts the service.
+  - *ue_version_major:* The version of the service.
+  - *resource_id:* The resource ID (value of 0 is used to indicate the default ID for the service).
+
+
+== Client APIs
```

Comment:
I am surprised to see the client facing API being described here again. Isn't the Client facing API described in the `client.adoc` file? Isn't that the whole purpose of the split-up of the original proto service into two separate (proto) services (`UDiscovery` and `UDiscoveryReplicator`)?